### DISCUSSION TITLE
Running crosstab query from Flutter?

### DISCUSSION THREAD

**Discussion by JulianSwales** [supabase/supabase#10285] [Questions] [answered]
So, how would I run this crosstab query in Flutter to supabase? 
select * from crosstab('select result_group,question_id,answer from survey_results where survey_id = 1 order by 1,2') as ct(row_name uuid, question1 jsonb, question2 jsonb,question3 jsonb,question4 jsonb,question5 jsonb,question6 jsonb,question7 jsonb)

**Comment by GaryAustin1** [comment, marked as answer]
For more complex SQL you have to write a Postgres function and call it with an rpc call from the client.  Sometime you can get close with an api call and finish up in the client by manipulating the returned data table. 
https://supabase.com/docs/reference/dart/rpc

**Reply by JulianSwales** [reply]
Thanks Gary, I was hoping to avoid having to use an rpc since the rpc requires a fixed typedef response, and these will be variable.. I guess I'll be resorting to an rpc with  json output and do the data manipulation on the client as you suggest..
